### PR TITLE
ci: Markdown linting

### PR DESCRIPTION
**Pull request by spydon**
# Description

Adds markdown linting to the CI and fixes up all the current files.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: doc/development/documentation.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ section will focus on the Markdown extensions that are enabled in our build syst
 
 The table of contents for the site must be created manually. This is done using special `{toctree}`
 blocks, one per each subdirectory:
+
+<!-- markdownlint-disable MD040 -->
```
Comment (st-pasha) [inline doc/development/documentation.md:25]: The language of this snippet (and below) is Markdown, we could specify that.

File: doc/development/style_guide.md [hunk 2]
```diff
@@ -201,6 +215,7 @@ Check the [Documentation] manual about how to work with the docs site.
 
 The following style rules generally apply when writing documentation (although we currently don't
 have a linter to enforce them strictly):
```
Comment (st-pasha) [inline doc/development/style_guide.md:217]: I guess this is no longer accurate :)

File: .markdownlint.yaml [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+# MD052/reference-links-images - Reference links and images should use a label that is defined
+MD052: true
+
+# MD053/link-image-reference-definitions - Link and image reference definitions should be needed
+MD053:
+  # Ignored definitions
+  ignored_definitions: [
+  "//"
+  ]
```
Comment (st-pasha) [inline .markdownlint.yaml:240]: nit: missing newline

File: packages/flame_flare/README.md [hunk 4]
```diff
@@ -2,12 +2,15 @@
...
 [![Discord](https://img.shields.io/discord/509714518008528896.svg)](https://discord.gg/pxrBmy4)
 
+
 # Flame Flare
 
-Integrate awesome [Flare animations](https://rive.app/explore/popular/trending/all) to your [Flame](https://flame-engine.org/) game.
+Integrate awesome [Flare animations](https://rive.app/explore/popular/trending/all) to your [Flame](https://flame-engine.org/)
+game.
 
 > :warning: **Flare is deprecated and no longer supported.** Please consider upgrading to Rive.
```
Comment (st-pasha) [inline packages/flame_flare/README.md:11]: This could be `{warning}` admonition.

**Review by st-pasha — APPROVED** [review]

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**